datafusion/core/src/execution/context/mod.rs

```diff
@@ -849,6 +849,33 @@ impl SessionContext {
         let table_paths = table_paths.to_urls()?;
         let session_config = self.copied_config();
         let listing_options = options.to_listing_options(&session_config);
+
+        let option_extention = listing_options.file_extension.clone();
+        let filename = table_paths[0].prefix().filename();
```

Comment:
we prob should check `table_paths` its not empty before accessing the element by index